app/jobs/confidence/days_homeless_job.rb

```diff
@@ -7,16 +7,84 @@
 module Confidence
   class DaysHomelessJob < BaseJob
     queue_as :long_running
-    include ArelHelper
 
+    # Grab a batch of queued census calculations
+    # and update at least on for each client.
     def initialize(client_ids:)
       @client_ids = client_ids
     end
 
+    private def dm_model
+      GrdaWarehouse::Confidence::DaysHomeless
+    end
+
     def perform
-      @client_ids.each do |id|
-        GrdaWarehouse::Confidence::DaysHomeless.calculate_queued_for_client(id)
+      # 1. get the first queued iteration for each client
+      queued_for_batch = dm_model.queued.where(
+        resource_id: @client_ids,
+      ).order(:resource_id, iteration: :asc).distinct_on(:resource_id)
+
+      unless queued_for_batch.any?
+        dm_model.logger.warn { "#{self.class} found no queued requests. Nothing to do." }
+        return
       end
+
+      # 2. get lists of dates for the provided client
+      # ids where they have only homeless services
+      # This is a batch version of GrdaWarehouse::Hud::Client.dates_homeless_scope
+      dates_by_client_id = {}
+      t = GrdaWarehouse::ServiceHistoryService.arel_table
+      GrdaWarehouse::ServiceHistoryService.where(
```

Comment:
If this is not limited to date range, it should query `GrdaWarehouse::ServiceHistoryServiceMaterialized`